src/bfd-disasm.cpp

```diff
@@ -1,49 +1,27 @@
 #include <cstdlib>
 #include <cstring>
-#include <fcntl.h>
+#include <fstream>
 #include <iostream>
-#include <sys/stat.h>
-#include <sys/types.h>
+#include <memory>
 #include <unistd.h>
 // bfd.h assumes everyone is using autotools and will error out unless
 // PACKAGE is defined. Some distros patch this check out.
 #define PACKAGE "bpftrace"
 #include "bfd-disasm.h"
+#include "utils.h"
 #include <bcc/bcc_elf.h>
 #include <bcc/bcc_syms.h>
 #include <bfd.h>
 #include <dis-asm.h>
 
 namespace bpftrace {
 
-BfdDisasm::BfdDisasm(std::string &path) : size_(0)
+BfdDisasm::BfdDisasm(std::string &path) : fs_{ path }, size_{ 0 }
 {
-  fd_ = open(path.c_str(), O_RDONLY);
```

Comment:
I strongly feel we should leave the FD based stuff in. This is an easily preventable TOCTOU bug waiting to happen that nobody enjoys debugging.